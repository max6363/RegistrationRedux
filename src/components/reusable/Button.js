import React, {Component} from 'react';
import {View, Text, TouchableOpacity} from 'react-native';

export default class Button extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    const {
      container,
      clickableStyle,
      textStyle,
      enabledTextColor,
      enabledBackgroundColor,
      disabledTextColor,
      disabledBackgroundColor,
    } = styles;
    const {title, isDisabled} = this.props;

    return (
      <View
        style={[
          container,
          isDisabled ? disabledBackgroundColor : enabledBackgroundColor,
        ]}>
        <TouchableOpacity
          disabled={isDisabled}
          style={[clickableStyle]}
          onPress={this.props.onPress}>
          <Text
            style={[
              textStyle,
              isDisabled ? disabledTextColor : enabledTextColor,
            ]}>
            {title}
          </Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = {
  container: {
    flex: 1,
    alignSelf: 'stretch',
    alignItems: 'center',
    justifyContent: 'center',
  },
  clickableStyle: {
    flex: 1,
    alignSelf: 'stretch',
    alignItems: 'center',
    justifyContent: 'center',
  },
  textStyle: {
    fontSize: 20,
    color: 'black',
  },
  enabledTextColor: {
    color: 'white',
  },
  enabledBackgroundColor: {
    backgroundColor: '#005493',
  },
  disabledTextColor: {
    color: 'white',
  },
  disabledBackgroundColor: {
    backgroundColor: '#00000050',
  },
};
