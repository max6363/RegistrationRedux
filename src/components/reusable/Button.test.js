import React from 'react';
import {shallow} from 'enzyme';
import Button from './Button';

describe('Test Button', () => {
  it('can test Button with title (Enabled)', () => {
    const wrapper = shallow(<Button title="Button" isDisabled={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('can test Button with title (Disabled)', () => {
    const wrapper = shallow(<Button title="Button" isDisabled={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
