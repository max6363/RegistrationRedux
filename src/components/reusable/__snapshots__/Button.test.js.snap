// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button can test Button with title (Disabled) 1`] = `ShallowWrapper {}`;

exports[`Test Button can test Button with title (Enabled) 1`] = `ShallowWrapper {}`;
